appcenter-analytics/ios/appcenter-analytics.podspec

```diff
@@ -1,6 +1,6 @@
 require 'json'
 
-package = JSON.parse(File.read(File.join(__dir__, 'package.json')))
+package = JSON.parse(File.read(File.join(__dir__, '../package.json')))
```

Comment:
We can't move the podspec location. Please see autolinking document at https://github.com/react-native-community/cli/blob/master/docs/autolinking.md#platform-ios:

> This means that all libraries need to ship a Podspec either in the root of their folder or where the Xcode project is. Podspec references the native code that your library depends on.

Reply:
Yes, I tested these changes with a blank RN 0.60 application, as well as with RN 0.59 and RN 0.55.
I think the auto-linking documentation is a bit confusing because it does not indicate that you can manually specify the path to the podspec file in the **react-native.config.js**. But it's in another [document](https://github.com/react-native-community/cli/blob/7350e0e86884d21cbd50a5c04e73f8482bfef07b/docs/dependencies.md#platformsiospodspecpath), which clearly states that this parameter can be used with the auto-linking.
The script specified in the [native_modules.rb](https://github.com/react-native-community/cli/blob/master/packages/platform-ios/native_modules.rb) documentation in [line 31](https://github.com/react-native-community/cli/blob/5d23c71f186e6d8739cfa57b378526f8146e9c0b/packages/platform-ios/native_modules.rb#L31) just reads it from the **react-native.config.js** file.
If we do not apply these changes, we will have to specify in the documentation that the **react-native link** does not work correctly, starting with SDK 2.2.0. I don’t think this issue has another proper fix.

BTW here is a sequence of commands that I have used for testing:

```
RN 0.60:
react-native init LinkDemo && cd LinkDemo

echo 'Removing existing appcenter* packages...'
rm -rf node_modules/appcenter*

echo "Packing appcenter* packages..."
npm pack ../appcenter
npm pack ../appcenter-analytics
npm pack ../appcenter-link-scripts

echo "Installing appcenter* packages..."
npm install appcenter*.tgz

echo "Cleanup appcenter*.tgz"
rm appcenter*.tgz

echo "Installing other packages..."
npm install

echo "Packing appcenter* packages..."
npm pack ../appcenter-auth

echo "Installing appcenter* packages..."
npm install appcenter*.tgz

echo "Cleanup appcenter*.tgz"
rm appcenter*.tgz

cd ios && pod install && cd ..

//do other setup, check that events are tracked in the portal.

RN 0.59:

react-native init LinkDemo --version react-native@0.59.8 && cd LinkDemo

echo 'Removing existing appcenter* packages...'
rm -rf node_modules/appcenter*

echo "Packing appcenter* packages..."
npm pack ../appcenter
npm pack ../appcenter-analytics
npm pack ../appcenter-link-scripts

echo "Installing appcenter* packages..."
npm install appcenter*.tgz

echo "Cleanup appcenter*.tgz"
rm appcenter*.tgz

echo "Installing other packages..."
npm install

react-native link appcenter
react-native link appcenter-analytics

echo "Packing appcenter* packages..."
npm pack ../appcenter-auth

echo "Installing appcenter* packages..."
npm install appcenter*.tgz

echo "Cleanup appcenter*.tgz"
rm appcenter*.tgz

react-native link appcenter-auth

//do other setup, check that events are tracked in the portal.

```

Reply:
That makes sense, thanks @russelarms for the clear explanation.